### DISCUSSION TITLE
Make GIF from png problems

### DISCUSSION THREAD

**Discussion by nguyennhatdung** [ImageMagick/ImageMagick#5563] [Help] [answered]
Hi
i'm trying make a gif image from png, howerver when i try do that on windows i got a problems : 
original image size was 130x130 but the gif result display is not 130x130 

`magick -dispose previous *.png -strip -coalesce -layers optimize -delay 2 -loop 0 -resize 130x130 out.gif`

in the end magick create 3205x145 gif  

i just want gif size equal to the png size. how to make it ?

Platform : Windows 11 x64, ImageMagick-7.1.0-Q16-HDRI
All src image in the attach
[src.zip](https://github.com/ImageMagick/ImageMagick/files/9562700/src.zip)

**Comment by fmw42** [comment, marked as answer]
Your PNG images have a large virtual canvas and you should not coalesce in that condition or it will use the virtual canvas as the size. That is why you are getting the large images. But use +repage and do not -layers optimize if you want exactly the 130x130 size. If you optimize, the images will be smaller.

`magick -dispose previous *.png +repage -set delay 10 -loop 0 out.gif`

**Reply by nguyennhatdung** [reply]
i just split the src use this command

`magick original.png -crop 22x1+0+0@ -quality 100 out.png`

this cause large virtual canvas ?

**Reply by nguyennhatdung** [reply]
+repage working however the result (exact 130x130) is somehow worse compare to the photoshop version. 
![problems](https://user-images.githubusercontent.com/52491730/190073453-9af3e605-c470-4607-a819-f4c004c6f106.png)

**Reply by fmw42** [reply]
worse in what way?  try adding -layers optimize and see if that helps

**Comment by snibgo** [comment]
@nguyennhatdung: I don't understnd why you are cropping or resizing. Your inputs are 130x130, and your want the GIF to be the same size.

Your PNG files have alpha varying from 0.0 (transparent) to 1.0 (opaque). GIF can record only binary alpha: each pixel is either fully transparent or fully opaque. What do you want to do about that?

If you want transparency in your GIF, then it will have to be binary, which will give jagged edges. Or you might flatten each frame against white, which will remove transparency, but give smooth edges.

**Reply by nguyennhatdung** [reply]
as mention b4 my PNG images have a large virtual canvas that's why i try to resize to the original size 130x130

GIF can record only binary alpha => thanks, i just convert png remove alpha and it seem nice now

`magick original.png -crop 22x1+0+0@ -quality 100 -background white -alpha remove -alpha off out.png
`
`magick -dispose previous *.png +repage -set delay 10 -loop 0 out.gif`
